packages/api-server/src/modules/assets.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  AssetGetRequest,
+  DendronError,
+  ERROR_STATUS,
+  RespV2,
+} from "@dendronhq/common-all";
+import { WorkspaceUtils } from "@dendronhq/engine-server";
+import { getWS } from "../utils";
+
+export class AssetsController {
+  static singleton?: AssetsController;
+
+  static instance() {
+    if (!AssetsController.singleton) {
+      AssetsController.singleton = new AssetsController();
+    }
+    return AssetsController.singleton;
+  }
+
+  async get({ fpath, ws }: AssetGetRequest): Promise<RespV2<string>> {
+    const engine = await getWS({ ws });
```

Comment:
not related to this review but - should we rename `getWS()` to `getDEngine()`?  I find it a bit confusing that this method returns a `DEngine` and not a `DWorkspace`

Also - getWS() takes a workspace root as a param. Do we expect the api-server to potentially have multiple `DEngine` instances? Currently, it seems 1-1, given that `WorkspaceController` is a singleton.

If the answer is no, then should we remove `ws` from being a parameter to this function?

Reply:
1. your right about the rename. it is confusing. will implement that in a separate review
2. the reason it takes a workspace is some pre-mature optimziation. the idea is in the future, instead of starting the workspace engine with vscode, the dendron engine will run as a daemon when you start up the computer. this will make it possible to integrate it with any running process (eg. chrome extension, alfred, other apps) as well as do indexing ahead of time instead of when launchig